test/common/shared_pool/shared_pool_test.cc

```diff
@@ -23,27 +27,47 @@ TEST(SharedPoolTest, Basic) {
 }
 
 TEST(SharedPoolTest, NonThreadSafeForGetObjectDeathTest) {
-  std::shared_ptr<ObjectSharedPool<int>> pool;
-
-  std::thread another_thread([&pool] { pool = std::make_shared<ObjectSharedPool<int>>(); });
-  another_thread.join();
+  Event::MockDispatcher dispatcher;
+  auto pool = std::make_shared<ObjectSharedPool<int>>(dispatcher);
+  pool->thread_id_ = std::thread::id();
+#ifdef NDEBUG
```

Comment:
Why is this necessary?

Reply:
release conditions will not trigger the assertions, even if not thread-safe can still run, this time will trigger the object's destructor to call `deleteObject` method, and then the internal call `disptaher: : post` method.

Reply:
The debug condition triggers the assertion directly, not the object destructor, so the call to `dispatcher::post` is not triggered.

Reply:
Let me see if I understand. In a release build, the ASSERT calls are no-ops and so in this test the getObject call succeeds and then the ObjectSharedPool is destroyed, which deletes the object.

In that case, maybe make the entire test `#ifndef NDEBUG` with an explanation of why it's not run in release?

Reply:
yup, your understanding is correct,  I'll add some comments to make the code easier to understand